### PR TITLE
Add mypy job to lint workflow

### PR DESCRIPTION
**Pull request by bveeramani**

### CODE REVIEW

**Comment by vinamrabenara** [comment]
Try rebasing your branch to resolve the confict.  and then do a force push. 

First do a ```git fetch upstream```
then ```git rebase -i upstream/master```
And then it will prompt you to resolve conflicts and commit the changes.

File: pylintrc [hunk 1]
```diff
@@ -42,6 +42,7 @@ disable=R,
         misplaced-comparison-constant,
         unsubscriptable-object,
         no-member,
+        consider-using-dict-items,
```
Comment (elibol) [inline pylintrc:45]: Same here. I think I made this change on a separate PR. Perhaps performing the rebase @vinamrabenara suggested will fix the PR.

File: nums/core/array/application.py [hunk 2]
```diff
@@ -234,7 +234,14 @@ def loadtxt(
     def scalar(self, value):
         return BlockArray.from_scalar(value, self.cm)
 
-    def array(self, array: np.ndarray, block_shape: tuple = None):
+    def array(self, array: Union[np.ndarray, List[float]], block_shape: tuple = None):
```
Comment (elibol) [inline nums/core/array/application.py:237]: These look like unrelated changes.

File: pylintrc (hunk 1 above)
Comment (vinamrabenara) [reply pylintrc:45]: This happened after i manually squashed some commits. This PR needs to be rebased.

File: nums/core/array/application.py (hunk 2 above)
Comment (bveeramani) [reply nums/core/array/application.py]: Fixed

File: pylintrc (hunk 1 above)
Comment (bveeramani) [reply pylintrc]: Fixed
